### DISCUSSION TITLE
Readings for the Nightscout website upbat plugin are duplicated.

### DISCUSSION THREAD

**Discussion by AgushkaNightscout** [NightscoutFoundation/xDrip#2428] [Q&A]
Hello. Thank you for this brilliant product! My battery reading in the NS is duplicated. In the database, the entries are like attached below. Upload to NS is done by AAPS and xDrip+, in xDrip+ all the checkboxes about sending the battery status are unchecked, so I assumed it's AAPS. But [MilosKozak](https://github.com/nightscout/AndroidAPS/issues/2163#issuecomment-1303231298) sad that only first one is AAPS, so second probably xDrip+. Can you please tell what should I switch off in xDrip+?
![Снимок экрана от 2022-11-03 23-46-45](https://user-images.githubusercontent.com/90107639/199956832-32cb42a4-e9a2-4935-bc86-d9ceddcd1df5.png)
![Снимок экрана от 2022-11-03 23-46-31](https://user-images.githubusercontent.com/90107639/199956840-6fe4dd16-b5af-4095-802b-6cb95711af09.png)

**Comment by Navid200** [comment]
Most people ran Nightscout on Heroku.  But, that's changing.
When you ask for help with Nightscout-related matters, it helps if you include what server your Nightscout are on.

**Reply by AgushkaNightscout** [reply]
Thank you for clarification. I used to ran NS on Heroku and had the same problem, well not a problem, just a nuisance. I don't think it should work this way, so I asked if this was normal behaviour for the programm. Now my NS 14.2.6 is on another hosting, here was readonly link

**Reply by Navid200** [reply]
I see that I did not clarify properly.
You could be running Nightscout on Heroku, on Google Cloud, or on other servers.  It helps to clarify where you are running your Nightscout on.

Does AAPS offer an option not to upload the battery data to Nightscout?

I personally use xDrip without AAPS.
Can you think of a scenario where someone uses AAPS without a CGM?

**Reply by AgushkaNightscout** [reply]
I have disabled the sending of the CGM transmitter battery, not the phone, in the xDrip+. Neither AAPS nor xDrip+ has the option to disable sending the battery status of the phone, as they can be used separately and not in conjunction with each other and the double readings in the upbat plugin on the NS website are normal.
Did i hear you right?
PS Could you please tell me if there is an option to check this in the future?

**Reply by Navid200** [reply]
We may have a misunderstanding.
I thought your question was how to stop xDrip from uploading the phone battery data to Nightscout.
I was never talking about the transmitter battery.
If yes, I provided my response.

If that's not your questions, please clarify what you are asking.

**Reply by AgushkaNightscout** [reply]
> I thought your question was how to stop xDrip from uploading the phone battery data to Nightscout.

Yes it was! But how I understood you - there is no way to stop xDrip to upload only battery status. You can turn on/off upload at all.

> I was never talking about the transmitter battery.

Yes, it was my mistake - (in the beginning) I thought, that I turned off upload phone battery, but this option even doesn't exist in xDrip, so all I turned off - is upload transmitter or bridge battery status.

so, the right question is - is it possible that in future there will be an option not to upload phone battery to NS?

**Reply by Navid200** [reply]
Have you asked that same question from the AAPS team?

**Reply by AgushkaNightscout** [reply]
[Yes](https://github.com/nightscout/AndroidAPS/issues/2163), but main discussion is here. xDrip's fine-tuning capabilities are greater.

**Reply by Navid200** [reply]
And as a result, we have an xDrip code that is extremely difficult and challenging to maintain or use.
People go through the enormous number of settings flipping them left and right and then post in a forum and say xDrip is not easy to use, you just install the Dexcom app and it works.

So, I don't want us to add yet one more setting unless it is absolutely necessary.
I have been uploading everything to my Nightscout for the past 4 years.  The database size is only 200MB.  So, I remove the battery content from it and after another 4 years, instead of 400MB, it will be 399MB.  Why is that even an issue?

**Reply by AgushkaNightscout** [reply]
I didn't mean to offend anyone, and if I did, I apologise. I am speaking from my point of view as a consumer, you as a developer are looking at the bigger picture. It seemed to me that the complexity of the program correlates strongly with the complexity of our general illness and sets us up for a fight and adding another switch does not make things much more complicated. After all, we're not iphone users.
I've already written that it's not a problem at all, just as a result of working in conjunction with AAPS on NS website - a visually annoying thing and it's not about the amount of space taken up.

Thank you for your hard work that helping, if not saving thousands of people!
